### DISCUSSION TITLE
Set Build Commands Question - add debug flag

### DISCUSSION THREAD

**Discussion by bjdarne** [geany/geany#3787] [Q&A]
I've been using Geany for a few days and have run into an issue that I'm unable to resolve.  I've created a test project and created a c application and compiled and linked it successfully.  I then decided to turn on debug flag and try out the Debug Plug-in.

I updated the compiler from **gcc -Wall -c "%f"** to **gcc -g -Wall -c "%f"** in both the Project Properties and the Set Build Commands windows.  After deleting the object and executable files, I compiled and linked successfully.  However, the Debug Plug-in didn't work.

To troubleshoot, I ran gdb in a terminal and noticed that the debug symbol information for my test application was missing.  After discovering this, I deleted the object and executable files and recompiled on the command line with the debug flag set (in the Geany folder).  After doing this gdb worked fine and I was able to run the Debug Plug-in within Geany.

It seems that the compiler flag I added to the property windows in Geany is not being picked up.  Note that the compiler message in the Compiler message window indicates that it is.

Any recommendation on what I may be missing/doing incorrectly?

Thanks.

**Comment by elextr** [comment]
To include debug info in the executable I think -g needs to be applied to the link stage as well as the compile stage.

**Reply by bjdarne** [reply]
You are correct!  It is working now.  Thank you very much!
